File: 13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_Integer = {
        'I': 1,
        "V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000
    }
        integerValue = 0
        for i, j in zip(s, s[1:]):
            if roman_Integer[i] < roman_Integer[j]:
                integerValue -= roman_Integer[i]
            else:
                integerValue += roman_Integer[i]

        return integerValue + roman_Integer[s[-1]]


        

File: 13-roman-to-integer/roman-to-integer.ts
function romanToInt(s: string): number {
    const romanInteger = {
    'I': 1,
    "V": 5,
    "X": 10,
    "L": 50,
    "C": 100,
    "D": 500,
    "M": 1000
  }
    let value = 0

    for (let i = 0; i + 1 < s.length; i++) {
        const letter = s[i]
        if (romanInteger[letter] < romanInteger[s[i + 1]]) {
        value = value - romanInteger[letter]
        } else {
        value = value + romanInteger[letter]
        }
    }
    value = value + romanInteger[s[s.length-1]]
    return value
};